File: Amazon/Q12.cpp
// { Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

 // } Driver Code Ends


class Solution{
    public:
    string colName (long long int n)
    {
        // your code here
        string result = "";
        int index = 0;
 
        while (n > 0){
 
            index = (n - 1) % 26;
	    // result += char(65 + index);
            result += ('A' + index);
            n = (n - 1) / 26;
        }
 
        reverse(result.begin(), result.end());
        return result;
    }
};

// { Driver Code Starts.
int main()
{
    int t; cin >> t;
    while (t--)
	{
		long long int n; cin >> n;
		Solution ob;
		cout << ob.colName (n) << '\n';
	}
}
  // } Driver Code Ends

File: Amazon/Q12.py
#User function Template for python3

class Solution:
    def colName (self, n):
        # your code here
        result = ""
 
        while n > 0:
 
        # find the index of the next letter and concatenate the letter
        # to the solution
 
        # here index 0 corresponds to `A`, and 25 corresponds to `Z`
            index = (n - 1) % 26
            result += chr(index + ord('A'))
            n = (n - 1) // 26
 
        return result[::-1]

#{ 
#  Driver Code Starts
#Initial Template for Python 3

t = int (input ())
for tc in range (t):
    n = int (input ())
    ob = Solution()
    print (ob.colName (n))
    

# } Driver Code Ends
